File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/completion/SimpleSnippetCompletionItem.java

package io.github.abc15018045126.sora.lang.completion;

import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;

import io.github.abc15018045126.sora.text.CharPosition;
import io.github.abc15018045126.sora.text.Content;
import io.github.abc15018045126.sora.widget.CodeEditor;

public class SimpleSnippetCompletionItem extends CompletionItem {

    private final SnippetDescription snippet;

    public SimpleSnippetCompletionItem(CharSequence label, SnippetDescription snippet) {
        this(label, null, snippet);
    }

    public SimpleSnippetCompletionItem(CharSequence label, CharSequence desc, SnippetDescription snippet) {
        this(label, desc, null, snippet);
    }

    public SimpleSnippetCompletionItem(CharSequence label, CharSequence desc, Drawable icon, SnippetDescription snippet) {
        super(label, desc, icon);
        this.snippet = snippet;
        kind(CompletionItemKind.Snippet);
    }



    @Override
    public void performCompletion(@NonNull CodeEditor editor, @NonNull Content text, @NonNull CharPosition position) {
        int prefixLength = snippet.getSelectedLength();
        var selectedText = text.subSequence(position.index - prefixLength, position.index).toString();
        int actionIndex = position.index;
        if (snippet.getDeleteSelected()) {
            text.delete(position.index - prefixLength, position.index);
            actionIndex -= prefixLength;
        }
        editor.getSnippetController().startSnippet(actionIndex, snippet.getSnippet(), selectedText);
    }

    @Override
    public void performCompletion(@NonNull CodeEditor editor, @NonNull Content text, int line, int column) {
        // do nothing
    }
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/completion/SimpleSnippetCompletionItem.kt
package io.github.abc15018045126.sora.lang.completion

import android.graphics.drawable.Drawable
import io.github.abc15018045126.sora.text.CharPosition
import io.github.abc15018045126.sora.text.Content
import io.github.abc15018045126.sora.widget.CodeEditor

class SimpleSnippetCompletionItem : CompletionItem {

    private val snippet: SnippetDescription

    constructor(label: CharSequence, snippet: SnippetDescription) : this(label, null, snippet)

    constructor(label: CharSequence, desc: CharSequence?, snippet: SnippetDescription) : this(label, desc, null, snippet)

    constructor(label: CharSequence, desc: CharSequence?, icon: Drawable?, snippet: SnippetDescription) : super(label, desc, icon) {
        this.snippet = snippet
        kind(CompletionItemKind.Snippet)
    }

    override fun performCompletion(editor: CodeEditor, text: Content, position: CharPosition) {
        val prefixLength = snippet.selectedLength
        val selectedText = text.subSequence(position.index - prefixLength, position.index).toString()
        var actionIndex = position.index
        if (snippet.deleteSelected) {
            text.delete(position.index - prefixLength, position.index)
            actionIndex -= prefixLength
        }
        editor.snippetController.startSnippet(actionIndex, snippet.snippet, selectedText)
    }

    override fun performCompletion(editor: CodeEditor, text: Content, line: Int, column: Int) {
        // do nothing
    }
}
